drivers: reset: rpi_pico: wait for reset deassertion

dsseng opened the pull request:
Failing to do so can lead to e.g. PIO not being ready by the time
a driver tries to use it.

pico-sdk also does block on this register in
its reset deassertion routines.

Signed-off-by: Dmitrii Sharshakov <d3dx12.xx@gmail.com>

soburi commented on drivers/reset/reset_rpi_pico.c:
```diff
@@ -106,7 +108,21 @@ static int reset_rpi_update(const struct device *dev, uint32_t id, uint8_t asser
+		if (ret) {
+			return ret;
+		}
+	} while (!assert && !(value & BIT(regbit)));
```
Is it okay not reflect the status of `config->active_low`?

dsseng committed:
drivers: reset: rpi_pico: rewrite

Use HAL functions, which also wait for reset to complete.

Remove unused register size and active-low DT props.

Signed-off-by: Dmitrii Sharshakov <d3dx12.xx@gmail.com>

soburi commented on drivers/reset/reset_rpi_pico.c:
```diff
@@ -1,122 +1,35 @@
 /*
- * Copyright (c) 2022 Andrei-Edward Popa
+ * Copyright (c) 2025 Dmitrii Sharshakov <d3dx12.xx@gmail.com>
```
Please do not erase the names of past author.

dsseng replied:
It's a full rewrite, so I thought the author of the original is not in charge of the current version. I do not object to keeping their copyright as well, if that is still applicable

soburi replied:
There are even functions with no diff at all, so I think that's enough to keep the original author's name.

dsseng replied:
Yea, agreed

soburi reviewed (approved):
LGTM

ajf58 reviewed (approved):
Sorry for taking so long to review this.
